### PR TITLE
Exclude optional params from flattening method args (discogapic)

### PR DESCRIPTION
**Pull request by andreamlin**

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/configgen/transformer/DiscoveryMethodTransformer.java [hunk 1]
```diff
@@ -17,32 +17,44 @@
...
   }
 
+  @Override
+  public boolean isIgnoredParameter(FieldModel parameter) {
+    Schema f = ((DiscoveryField) parameter).getDiscoveryField();
+    return PAGING_PARAMETERS.getIgnoredParameters().contains(parameter.getSimpleName())
+        || !(parameter.isRequired()
+            || f.isPathParam()
+            || Strings.isNullOrEmpty(f.description())
+            || !f.description().toLowerCase().contains("optional"));
```
[the remark this answers is not in the document]
Comment (pongad) [reply src/main/java/com/google/api/codegen/configgen/transformer/DiscoveryMethodTransformer.java]: I'm probably missing something, but shouldn't this just be `!parameter.isRequired()`? I think the API is doing something funny if the discovery doc has `"required": true` but the field is optional.

[the remark this answers is not in the document]
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/configgen/transformer/DiscoveryMethodTransformer.java]: Huh that's a very good point. Justifications for still using that boolean:
1. It's the last case in a long series of booleans, so lazy evaluation will hopefully determine that a field is required before we get to this "contains 'optional'" boolean.
2. It hasn't failed me so far in Compute. :) I might get burned later on.
3. There are some query parameters, such as "address" in method compute.addresses.patch, that are required, but there's no way to determine that from just looking at that schema in the discovery doc. Switching on "optional" in the description distinguishes those required query parameters from optional query params.

[the remark this answers is not in the document]
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/configgen/transformer/DiscoveryMethodTransformer.java]: @pongad Oh yes indeed. Will fix that haha

**Pull request merged**